### PR TITLE
Generate valid xcconfig when target includes spaces

### LINKED ISSUE #4783: inherit! :search_paths creates invalid xcconfig
With a Podfile like this:

    target "My App" do
        pod "A"
        target "Tests" do
            inherit! :search_paths
            pod "B"
        end
    end
    
    target "WatchExt" do
        pod "A"
    end

cocoapods-1.0.0.beta.2 creates an xcconfig for the Tests target like:

    FRAMEWORK_SEARCH_PATHS = $(inherited) $(PLATFORM_DIR)/Developer/Library/Frameworks "$PODS_FRAMEWORK_BUILD_PATH"
    ...
    PODS_FRAMEWORK_BUILD_PATH = $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-Tests $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-My App

Which is incorrect (it creates one framework search path pointing at a space-separated concatenation of paths, which is generally invalid but particularly bad when one of the paths has a space)

It should instead create:

    FRAMEWORK_SEARCH_PATHS = $(inherited) $(PLATFORM_DIR)/Developer/Library/Frameworks $PODS_FRAMEWORK_BUILD_PATH
    ...
    PODS_FRAMEWORK_BUILD_PATH = "$(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-Tests" "$(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-My App"

(Note lack of quotes in FRAMEWORK_SEARCH_PATHS & addition of per-path quotes in PODS_FRAMEWORK_BUILD_PATH)

Unfortunately, I can't upload my actual Podfile 'cos it's full of proprietary stuff... hopefully I've regressed this correctly :(

### PR DESCRIPTION
**Pull request by dnkoutso**
closes #4783.

/cc @segiddins @OneSadCookie

### CODE REVIEW

**Comment by dnkoutso** [comment]
Seems like the existing integration specs provide sufficient coverage for this so I didnt add a new test, let me know.

File: CHANGELOG.md [hunk 1]
```diff
@@ -651,6 +651,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Generate valid xcconfig when target name includes spaces.
```
Comment (segiddins) [inline CHANGELOG.md]: missing the trailing two spaces

**Comment by segiddins** [comment]
:+1: other than the changelog comment, thanks @dnkoutso !

File: CHANGELOG.md (hunk 1 above)
Comment (dnkoutso) [inline CHANGELOG.md]: whoa this was in the complete wrong section too of 0.38.2...fixing.

Comment (dnkoutso) [inline CHANGELOG.md]: @segiddins I am not seeing what you mean with missing trailing two spaces...

I fixed the location of the changelog though for my fix...

**Comment by dnkoutso** [comment]
Thank you! I will update the submodule SHA when the other PR is merged.

File: CHANGELOG.md (hunk 1 above)
Comment (segiddins) [inline CHANGELOG.md]: the sentence part of the changelog entry should have two trailing spaces after `.` to force a line break

**Comment by segiddins** [comment]
@dnkoutso no need to update the submodule SHA, I'll just hit restart on CI

File: CHANGELOG.md (hunk 1 above)
Comment (dnkoutso) [inline CHANGELOG.md]: Duh. Fixed the one below too by @efirestone

**Comment by segiddins** [comment]
Ace! Will merge once CI runs. Thanks @dnkoutso !

**Comment by dnkoutso** [comment]
Seems like I missed some in the other project with `FRAMEWORK_SEARCH_PATHS`...fixing.

**Comment by dnkoutso** [comment]
```Bacon::Error: File comparison error `Pods/Target Support Files/Pods-OS X App-PodTest/Pods-OS X App-PodTest.xcconfig` for install_subspecs:
--- DIFF -----------------------------------------------------------------------
 CODE_SIGN_IDENTITY =
 CONFIGURATION_BUILD_DIR = $PODS_FRAMEWORK_BUILD_PATH
-FRAMEWORK_SEARCH_PATHS = $(inherited) $PODS_FRAMEWORK_BUILD_PATH
+FRAMEWORK_SEARCH_PATHS = $(inherited) "$PODS_FRAMEWORK_BUILD_PATH"
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
 HEADER_SEARCH_PATHS = "${PODS_ROOT}/Headers/Private" "${PODS_ROOT}/Headers/Public"
-PODS_FRAMEWORK_BUILD_PATH = "$(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-OS X App"
+PODS_FRAMEWORK_BUILD_PATH = $(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/Pods-OS X App
 PODS_ROOT = ${SRCROOT}
 PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.${PRODUCT_NAME:rfc1034identifier}
 SKIP_INSTALL = YES
--- END ------------------------------------------------------------------------```

I might be missing another spot to make this change.

**Comment by segiddins** [comment]
Did you update the integration specs to include https://github.com/CocoaPods/cocoapods-integration-specs/pull/57/files ?

**Comment by dnkoutso** [comment]
Ah thats it. Hold up push coming shortly.

**Comment by dnkoutso** [comment]
Nope I am lying, the commit I am using is the latest one including the latest changes but there is still one test failing...

**Comment by dnkoutso** [comment]
@segiddins I think:

```ruby
          if target.requires_frameworks? && target.scoped?
            # Only quote the FRAMEWORK_SEARCH_PATHS entry, because it’s a setting that takes multiple values.
            # In addition, quoting CONFIGURATION_BUILD_DIR would make it be interpreted as a relative path.
            build_settings = {
              'PODS_FRAMEWORK_BUILD_PATH' => target.configuration_build_dir,
              'FRAMEWORK_SEARCH_PATHS' => '"$PODS_FRAMEWORK_BUILD_PATH"',
              'CONFIGURATION_BUILD_DIR' => '$PODS_FRAMEWORK_BUILD_PATH',
            }
            @xcconfig.merge!(build_settings)
          end
```

Also needs updating in `pod_xcconfig.rb`

**Comment by segiddins** [comment]
@dnkoutso yup, that's it

**Comment by dnkoutso** [comment]
Green!

**Comment by OneSadCookie** [comment]
Thanks so much for picking this up :)

**Pull request merged**